optimum/habana/transformers/models/mistral/modeling_mistral.py

```diff
@@ -386,28 +389,35 @@ def forward(
                     )
             else:
                 # first token
-                if flash_attention_causal_mask:
-                    # causal masking on first token requires inputs to be of the same length
-                    with ht.sdp_kernel(enable_recompute=flash_attention_recompute):
-                        attn_output = self.fused_scaled_dot_product_attention(
-                            query_states, key_states, value_states, None, 0.0, True, None
-                        )
+                if (
+                    not self.training
+                    and q_len == key_states.size(-2)
+                    and q_len > 8192
+                    and os.getenv("QUANT_CONFIG", "")
+                ):
+                    htcore.mark_step()
+                    attn_output = self.fused_sdpa_long(
```

Comment:
Do we still need this special handling for the longer sequence? Last time when I tried as long as the use_flash_attention works, that showed better performance. wondering if we still need this special handling for the longer q_len.